Library/Homebrew/extend/pathname.rb

```diff
@@ -299,11 +299,15 @@ def sha256
     Digest::SHA256.file(self).hexdigest
   end
 
-  sig { params(expected: T.nilable(Checksum)).void }
+  sig { params(expected: T.nilable(T.any(Checksum, BottleChecksum))).void }
   def verify_checksum(expected)
     raise ChecksumMissingError if expected.blank?
 
-    actual = Checksum.new(expected.hash_type, send(expected.hash_type).downcase)
+    actual = if expected.is_a? BottleChecksum
```

Comment:
This seems like a smell to me to check the types explicitly.

Reply:
I have trouble understanding what this `verify_checksum` method is doing, I need to investigate this a little bit more.
It compares an expected checksum to an actual checksum: but the actual checksum is just a copy of the expected one, so this test returns always true in my opinion ...

Reply:
> It compares an expected checksum to an actual checksum: but the actual checksum is just a copy of the expected one, so this test returns always true in my opinion ...

This isn't quite the case (but the code isn't super easy to follow).

```ruby
Checksum.new(expected.hash_type, send(expected.hash_type).downcase)
```

that's saying: create a new actual `Checksum` with the same hash type (e.g. `sha256`) as the expected checksum but the value is set from `send(expected.hash_type)` i.e. `send(:sha256)` i.e. call the `sha256` on the current `Pathname` object to get the actual checksum value.

Reply:
As we are only comparing the hexdigest, this is not needed anymore. Checksum is enough.